File: 2348.cpp
#include <assert.h>
#include <iostream>
#include <vector>
using namespace std;
class Solution {
public:
  long long zeroFilledSubarray(vector<int> &nums) {
    long long curr = 0;
    long long sum = 0;
    for (int i = 0; i < nums.size(); ++i) {
      if (nums[i] == 0) {
        ++curr;
        sum += curr;
      } else {
        curr = 0;
      }
    }
    return sum;
  }
};

int main() {
  vector<int> t1 = {1, 3, 0, 0, 2, 0, 0, 4};
  assert(Solution().zeroFilledSubarray(t1) == 6);
  vector<int> t2 = {0, 0, 0, 2, 0, 0};
  assert(Solution().zeroFilledSubarray(t2) == 9);
}

File: 2348.py
class Solution:
    def zeroFilledSubarray(self, nums: list[int]) -> int:
        currLength = 0
        sum = 0
        for i in range(len(nums)):
            if nums[i] == 0:
                currLength += 1
                sum += currLength
            elif nums[i] != 0 and currLength != 0:
                currLength = 0
        return sum


s = Solution()
assert s.zeroFilledSubarray([1, 3, 0, 0, 2, 0, 0, 4]) == 6
assert s.zeroFilledSubarray([0, 0, 0, 2, 0, 0]) == 9
